rust/benches/array_from_vec.rs

```diff
@@ -23,13 +23,19 @@ use criterion::Criterion;
 extern crate arrow;
 
 use arrow::array::*;
+use arrow::array_data::ArrayDataBuilder;
+use arrow::buffer::Buffer;
+use arrow::datatypes::*;
 
 fn array_from_vec(n: usize) {
-    let mut v: Vec<i32> = Vec::with_capacity(n);
+    let mut v: Vec<u8> = Vec::with_capacity(n);
     for i in 0..n {
-        v.push(i as i32);
+        v.push((i % 0xffff) as u8);
```

Comment:
I think this should be `i & 0xfff`. Yes, technically, this is the same for positive integers (which is ensured since `n: usize` and therefore `i: usize`), but modulo operator is a) usually slower (might be implemented using `DIV`) and b) cannot be optimized that well.

See https://godbolt.org/g/bL2jJE (debug build)
and https://godbolt.org/g/bQi1Zs (release build)

Reply:
Yes, I've meant `i & 0xffff`. Sorry.